src/accelerate/data_loader.py

```diff
@@ -1155,11 +1149,16 @@ def prepare_data_loader(
 
     new_dataset = dataloader.dataset
     # Iterable dataset doesn't like batch_sampler, but data_loader creates a default one for it
+    if isinstance(dataloader.sampler, BatchSampler):
+        raise ValueError(
+            "Should not pass a BatchSampler do dataloader sampler argument. As per pytorch>2.1.0 documentation, please pass this to sampler instead"
+        )
```

Comment:
Can you please show the exact paragraph/section of the pytorch docs that state this? You reference many issues, but I dont' see where you found this

Reply:
Thanks for the quick reply! Yes it’s referenced under motivation #5. 

“Official PyTorch Documentation for Reference” showing two arguments for sampler and batch_sampler,

And in the docs they write this is “mutually exclusive with sampler” 

`batch_sampler (Sampler or Iterable, optional) – like sampler, but returns a batch of indices at a time. Mutually exclusive with batch_size, shuffle, sampler, and drop_last.`

Reply:
If we pass a BatchSampler (yields a list of indices) as an argument to sampler, we will get a list of list of indices, because pytorch will convert whatever was passed in the sampler argument to “batch mode”.